File: uoo6uoow/Week12/스택.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();

        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            String input = scanner.nextLine();
            if (input.equals("top")) {
                if (!stack.isEmpty()){
                    System.out.println(stack.peek());
                }else {
                    System.out.println(-1);
                }

            } else if (input.equals("size")) {
                System.out.println(stack.size());

            } else if (input.equals("empty")) {
                if (stack.isEmpty()){
                    System.out.println(1);
                }else System.out.println(0);

            } else if (input.equals("pop")) {
                if (!stack.isEmpty()){
                    System.out.println(stack.pop());
                }else {
                    System.out.println(-1);
                }
            }else if (input.startsWith("push")){
                String[] com = input.split(" ");
                int num = Integer.parseInt(com[1]);
                stack.push(num);
            }
        }
    }
}

File: noeyeyh/Week12/스택.py
import sys
input = sys.stdin.readline

n = int(input())
stack = []
size = 0
 
for i in range(n):
    l = list(input().split())

    if l[0] == "push":
        stack.append(l[1])
        size += 1
    
    elif l[0] == "pop":
        if size != 0:
            print(stack.pop())
            size -= 1
        else:
            print(-1)

    elif l[0] == "size":
        print(size)

    elif l[0] == "empty":
        if size:
            print(0)
        else:
            print(1)

    else:
        if size != 0:
            print(stack[-1])
        else:
            print(-1)